app/services/project_with_domain_updater.rb

```diff
@@ -0,0 +1,49 @@
+require 'resolv'
+
+class ProjectWithDomainUpdater
+  def self.get_cname(domain)
+    Rails.logger.info("Getting CNAME for #{domain}")
+
+    begin
+      r =
+        Resolv::DNS.open do |dns|
+          dns.getresource(domain, Resolv::DNS::Resource::IN::CNAME)
+        end
+      r = r.name.to_s
+      Rails.logger.info("CNAME = #{r.inspect}")
+      r
+    rescue Resolv::ResolvError => e
+      Rails.logger.info("Error: #{e}")
+      nil
+    end
+  end
+
+  def initialize(request:, project:)
+    @request = request
```

Comment:
и не передавать его потом отдельно

```suggestion
    @request = request
    @domain = @request.params[:project][:domain]
```

Reply:
Выпилю оттуда `request`, чтобы не смущал :)